pkg/tcpip/link/rawfile/blockingpoll_amd64.s

```diff
@@ -14,27 +14,28 @@
 
 #include "textflag.h"
 
-// BlockingPoll makes the poll() syscall while calling the version of
+// BlockingPpoll makes the ppoll() syscall while calling the version of
 // entersyscall that relinquishes the P so that other Gs can run. This is meant
 // to be called in cases when the syscall is expected to block.
 //
-// func BlockingPoll(fds *PollEvent, nfds int, timeout int64) (n int, err syscall.Errno)
-TEXT ·BlockingPoll(SB),NOSPLIT,$0-40
+// func BlockingPpoll(fds *PollEvent, nfds int, timeout *timespec, sigmask *sigset) (n int, err syscall.Errno)
```

Comment:
I think you should omit the sigmask argument since it is unused. You could also keep the name as BlockingPoll if you'd like as well. The use of ppoll isn't too important.

Reply:
So you suggest to use 
func BlockingPoll(fds *PollEvent, nfds int, timeout *timespec) (n int, err syscall.Errno)
as the signature, and still call the SYS_PPOLL in the body?

Reply:
fixed!